lib/MusicBrainz/Server/Authentication/Store.pm

```diff
@@ -43,7 +43,9 @@ sub for_session
 sub from_session
 {
     my ($self, $c, $frozen) = @_;
-    return _rebless_editor($c->model('Editor')->get_by_id($frozen->{id}));
+    my $editor = $c->model('Editor')->get_by_id($frozen->{id});
+    return unless ($editor && !$editor->deleted);
+    return _rebless_editor($editor);
```

Comment:
This is the main part of this PR, fixing the issue. `from_session` is called with the session data in `$frozen` when a browser has sent a cookie with a valid session ID, and the user returned here is immediately accepted. This avoids all the checks that were added in https://bitbucket.org/metabrainz/musicbrainz-server/pull-requests/448/mbs-6182-delete-editors-with-a-dedicated